character.cpp

```diff
@@ -1,22 +1,51 @@
 #include "character.h"
 #include <fstream>
 
-Character::Character(const std::string& name, int hp, const int dmg) :Name(name), HP(hp), DMG(dmg) {}
+//konstruktor
+Character::Character(const std::string& name, int hp, int dmg) :Name(name), HP(hp), DMG(dmg) {
+	Character::MaxHP = HP;
+	Character::XP = 0;
+}
 
+//getter
 const std::string& Character::getName() const {
 	return Character::Name;
 }
-void Character::attackEnemy(Character& enemy) const {
+
+//szintnoveles
+//eletero es tamadoero a 10%-kal no
+//maximalis eletero az megvaltozott eletero erteket veszi fel
+//XP-t 100-zal csokkentjuk
+double Character::levelUp() {
+	int lvl = Character::Level - 1;
+	Character::HP = Character::MaxHP * 1.1 * (lvl);
+	Character::DMG = Character::DMG * 1.1 * (lvl);
+	Character::MaxHP = Character::HP;
+	Character::Level++;
```

Comment:
Egyrészt nem kellenek a scope-ok, a metóduson belül amúgy is scope-ban vannak a tagváltozók.
Másrészt hiányzik a kerekítés a növelésnél.
Harmadrészt nem jó a képlet, mert ha be van szorozva (level-1)-gyel, akkor ott elég nagy szorzókkal lesz beszorozva, duplázódni, triplázódni, stb. fog, és nem csak 10%-kal fog nőni. De az első szintlépésnél meg 0-val szorzódik, az meg kinullázza a maxHP-t és a sebzést, nem?